### DISCUSSION TITLE
How to profile snippets?

### DISCUSSION THREAD

**Discussion by MattiaMontanari** [NVIDIA-Omniverse/PhysX#120] [Q&A]
I'm on Windows and I've build PhysX with the profile configuration. I noticed there are few Snippets I can play with, including a custom profiling. I run it, I see the various entries in the terminal, but I get not data. Is it possible to extract wall-time data and visualise them? If so, how?

**Comment by preist-nvidia** [comment]
Hi @MattiaMontanari - you can try using PVD2 or PVD3 for profiling, but we do not support them anymore. We are aware that we need to provide an updated profiling solution but I can't provide a timeline at this point.

**Comment by MattiaMontanari** [comment]
Hi @preist-nvidia, got it: I tried PVD 3 (2021.01.29532135). The exe built from PhysX5 Snippets do connect nicely to PVD, however when I export the clip PVD crashes. If I try to open the profile tab after few seconds of the  `SnippetCustomProfiler` or `SnippetCustomGeometryCollision` PVD freezes. 
Do you have a workaround to create a profile view (such as [this one](https://user-images.githubusercontent.com/42665265/184474594-46d11497-901d-4f79-9f50-bbfcffb412bf.png) from [this recent issue](https://github.com/NVIDIAGameWorks/PhysX/issues/593)) with PhysX 5?

**Reply by preist-nvidia** [reply]
You could try logging only profile info, so change `PxPvdInstrumentationFlag::eALL` in the snippet setup to `PxPvdInstrumentationFlag::ePROFILE` and see if that works.

**Reply by MattiaMontanari** [reply]
No luck. After a clean build in Profile config with `::ePROFILE`, PVD 3 still freezes when trying to open the profile tab. 

How do you profile the engine? Do you also use a 3rd party tool?

**Comment by Zeblote** [comment]
I've been using Superluminal profiler on release builds. Accumulating samples over 10-20 seconds of frames gives a much more accurate picture of actual performance, especially with functions that get called many times in a loop, than instrumentation based profiling ever could. You'll also get insights on specifically which part of the code is slow, rather than just a high level overview.

**Reply by MattiaMontanari** [reply]
Thanks! I was hoping to get a simpler/native profiler (PVD is doesn't look *that* bad at the end of the day), but I did not know about Superluminal. I'll give it a go!

**Reply by Zeblote** [reply]
It's the nicest and least clunky sampling profiler I've ever used. So fast, too. Really fun to use.
